### PR TITLE
Add reflection config for Storage APIs

### PR DESCRIPTION
**Pull request by dzou**
Adds preliminary support for the Storage APIs. 

Going to use this `org.graalvm.nativeimage.hosted.Feature` which is a way of declaring reflection configuration via Java code instead of JSON files. The benefit of this will be that we can organize things and also only add reflective classes if their library is present on the classpath (see `StorageFeature`).

### CODE REVIEW

File: .cloudbuild/graal-build.yaml [hunk 1]
```diff
@@ -1,7 +1,14 @@
 steps:
-  - name: oracle/graalvm-ce:20.2.0-java11
+  - name: oracle/graalvm-ce:20.1.0-java11
```
Comment (ttomsu) [inline .cloudbuild/graal-build.yaml:2]: Downgrading?

File: google-cloud-graalvm-samples/storage-sample/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Pub/Sub Sample Application with GraalVM
```
Comment (ttomsu) [inline google-cloud-graalvm-samples/storage-sample/README.md:1]: s/Pub.Sub/Stoage/g

File: google-cloud-graalvm-support/pom.xml [hunk 3]
```diff
@@ -11,4 +11,11 @@
...
   <artifactId>google-cloud-graalvm-support</artifactId>
 
+  <dependencies>
+    <dependency>
+      <groupId>org.graalvm.nativeimage</groupId>
+      <artifactId>svm</artifactId>
+      <version>${graalvm.version}</version>
+    </dependency>
+  </dependencies>
 </project>
```
Comment (ttomsu) [inline google-cloud-graalvm-support/pom.xml:21]: nit: newline

**Review by ttomsu — APPROVED** [review]

File: google-cloud-graalvm-samples/storage-sample/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+The Pub/Sub sample application demonstrates some common operations with Pub/Sub and is compatible with GraalVM compilation.
```
Comment (elefeint) [inline google-cloud-graalvm-samples/storage-sample/README.md:3]: Another couple of "Pub/Sub"s got in.

**Review by elefeint — CHANGES_REQUESTED** [review]
Looks great; structured code is much easier on the eyes than a wall of config (although I'll admit, I wondered if we should not introduce Spring Boot style configuration with our own `@ConditionalOnClass` and a list of classes to register for reflection etc. But that would likely bring us back to the wall of config...)

Same question as Travis about downgrading dependency.

File: .cloudbuild/graal-build.yaml (hunk 1 above)
Comment (dzou) [reply .cloudbuild/graal-build.yaml:2]: Ah yeah, I was getting this error using version 20.2.0:

```
Step #0: Error: Already registered: java.util.zip.ZipFile$CleanableResource.get(ZipFile, File, int)
Step #0: com.oracle.svm.core.util.UserError$UserException: Already registered: java.util.zip.ZipFile$CleanableResource.get(ZipFile, File, int)
Step #0: 	at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
Step #0: 	at com.oracle.svm.core.util.UserError.guarantee(UserError.java:92)
Step #0: 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.register(AnnotationSubstitutionProcessor.java:724)
```

I read on some Slack threads that this is an issue with 20.2.0; not sure how to resolve it just yet but the workaround solution they said was just to use `20.1.0` for now. I'll upgrade to `20.2.0` in a separate PR.

File: google-cloud-graalvm-samples/storage-sample/README.md (hunk 2 above)
Comment (dzou) [reply google-cloud-graalvm-samples/storage-sample/README.md]: Done.

File: google-cloud-graalvm-support/pom.xml (hunk 3 above)
Comment (dzou) [reply google-cloud-graalvm-support/pom.xml]: Done. Sorry I forgot to update that checkstyle to check for new lines at end of files; will send PR out shortly.

File: google-cloud-graalvm-samples/storage-sample/README.md (hunk 4 above)
Comment (dzou) [reply google-cloud-graalvm-samples/storage-sample/README.md]: Done.

**Comment by dzou** [comment]
> Looks great; structured code is much easier on the eyes than a wall of config (although I'll admit, I wondered if we should not introduce Spring Boot style configuration with our own `@ConditionalOnClass` and a list of classes to register for reflection etc. But that would likely bring us back to the wall of config...)
> 
> Same question as Travis about downgrading dependency.

Yeah the version downgrade is a temporary workaround until I figure out how to fix the error associated with `20.2.0` that I mentioned.

**Review by elefeint — APPROVED** [review]

**Review by meltsufin — APPROVED** [review]
Just realized I reviewed it a while ago and forgot to submit, but all of my comments already got addressed. 😄 
LGTM!

**Pull request merged**